core/base/scalarFieldCriticalPoints/ScalarFieldCriticalPoints.h

```diff
@@ -27,6 +27,7 @@
 
 #pragma once
 
+#include <algorithm>
```

Comment:
why is the `<algorithm>` needed here?
(I don't see any reason)

if it builds and runs without it, thanks for removing it.